src/ImageProcessor.Web/Processors/OverlayColor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="OverlayColor.cs" company="James Jackson-South">
+//   Copyright (c) James Jackson-South.
+//   Licensed under the Apache License, Version 2.0.
+// </copyright>
+// <summary>
+//   Adds a color overlay to the current image.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ImageProcessor.Web.Processors
+{
+	using System.Collections.Specialized;
+	using System.Drawing;
+	using System.Text.RegularExpressions;
+	using System.Web;
+
+	using ImageProcessor.Processors;
+	using ImageProcessor.Web.Helpers;
+
+	/// <summary>
+	/// Adds a color overlay to the current image.
+	/// </summary>
+	public class OverlayColor : IWebGraphicsProcessor
+	{
+		/// <summary>
+		/// The regular expression to search strings for.
+		/// </summary>
+		private static readonly Regex QueryRegex = new Regex("color=[^&]", RegexOptions.Compiled);
```

Comment:
I'm not sure this Regex overlaps with the `bgcolor=` query, but while testing I haven't seen any errors indicating this would be an issue.